### ISSUE TITLE
Incorrect type evaluated for `TypeIs[type[T]]`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7946]
The narrowed type in this case is incorrect:

```python
from dataclasses import Field
from typing import Any, ClassVar, Protocol, runtime_checkable
from typing_extensions import TypeIs

@runtime_checkable
class DataclassInstance(Protocol):
    __dataclass_fields__: ClassVar[dict[str, Field[Any]]]

def is_dataclass(obj: type) -> TypeIs[type[DataclassInstance]]: ...

def test1(x: type) -> None:
    if is_dataclass(x):
        reveal_type(x)  # Should be type[DataclassInstance]
```

**Comment by erictraut** [comment]
This is related to https://github.com/python/typeshed/pull/11929.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.364.
